lua/lvim/core/treesitter.lua

```diff
@@ -13,7 +13,37 @@ M.config = function()
     highlight = {
       enable = true, -- false will disable the whole extension
       additional_vim_regex_highlighting = false,
-      disable = { "latex" },
+      disable = function(lang, buf)
+        if vim.tbl_contains({ "latex" }, lang) then
+          return true
+        end
+
+        local max_filesize = 1024 * 1024
+        local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(buf))
+        if ok and stats and stats.size > max_filesize then
+          if lvim.builtin.illuminate.active then
+            pcall(require("illuminate").pause_buf)
+          end
+
+          if vim.tbl_contains({ "json" }, lang) then
+            vim.schedule(function()
+              vim.api.nvim_buf_call(buf, function()
+                vim.cmd "NoMatchParen"
+                vim.cmd "syntax clear"
```

Comment:
> is there a setting to disable `list[char]`?

I have no idea

Reply:
there is! we can also add a few more that we're originally setting
```lua
    vim.cmd "setlocal nocursorline nolist noundofile bufhidden=unload"
```

Reply:
> there is! we can also add a few more that we're originally setting
> 
> ```lua
>     vim.cmd "setlocal nocursorline nolist noundofile bufhidden=unload"
> ```

Added these and also `syntax off` because sometimes `pattern uses more memory than 'maxmempattern'` would come up when changing buffers